sync/bug.py

```diff
@@ -130,9 +130,9 @@ def comment(self, bug_id, comment, **kwargs):
                               method='POST', json=body)
 
     def new(self, summary, comment, product, component, whiteboard=None, priority=None,
-            url=None):
+            url=None, bug_type="task"):
```

Comment:
when the list of arguments start to be long it might be time to start thinking of passing another type of object, such as a dictionary. to think about the future.

Reply:
I'm generally not a fan of dictionaries as arguments just as a way to package some fields together; it has a few disadvantages compared to keyword arguments:
* Reduces documentation from the function signature since it ends up with a generic `options` element.
* Harder to spot typos, since they don't end up with `InvalidArgumentError` but something at the point of use
* Often requires more code to unpack the dictionary.

That said I agree there are some advantages. I think Py3's keyword-only arguments probably address most of the problems, so once we do the Py3 conversion we can probably improve things. And for really egregious cases you can always make a custom object type like you would in a language with a proper type system.